include/boost/static_string/static_string.hpp

```diff
@@ -729,6 +729,57 @@ defined(BOOST_STATIC_STRING_NO_PTR_COMP_FUNCTIONS)
     std::less<const T*>()(ptr, src_last);
 #endif
 }
+
+// This nested type workaround is for gcc 5,
+// which prohibits throw expressions in constexpr
+// functions, but for some reason permits them in
+// constructors.
+template<typename Exception>
+struct exception
+{
+#ifdef BOOST_STATIC_STRING_GCC5_BAD_CONSTEXPR
+  struct raise_if
+  {
+    BOOST_STATIC_STRING_CPP14_CONSTEXPR
+    raise_if(
+      bool check, 
+      const char* message)
+    {
+      BOOST_STATIC_STRING_THROW_IF(
```

Comment:
This looks like a lot of unnecessary complexity why are we doing this bool check and the THROW_IF and all of that?

The `bool check` is going to make coverage reports less reliable because you are hiding the condition in the macro.

Reply:
I suppose we can get rid of `THROW_IF` now that we only use it in a single place

Reply:
For example look here:
https://codecov.io/gh/sdkrystian/static_string/src/01996796a0789c1a4c1ebe116853c8f250a6c1bf/include/boost/static_string/static_string.hpp#L1609

How do we know that we are getting coverage for both cases, where `pos >= size()` and `pos < size()`? The report doesn't show it, because you have hidden it in a macro behind a member function behind a class template. The function `at` should be written thusly:
```
#ifndef BOOST_STATIC_STRING_DOCS
namespace detail {
struct out_of_range_error : std::out_of_range {
  static void raise(char const* s) {
    BOOST_THROW_EXCEPTION( detail::out_of_range_error(s) );
  }
} // detail
#endif

BOOST_STATIC_STRING_CPP14_CONSTEXPR
reference
at(size_type pos)
{
  if( pos >= size() )
    detail::out_of_range_error::raise();
  return data()[pos];
}
```